Life4/View Controllers/LoginViewController.swift

```diff
@@ -11,16 +11,66 @@
 import UIKit
 
 class LoginViewController: UIViewController {
-
+    static let userNameNotification = Notification.Name("userNameNotification")
+    static let rivalCodeNotification = Notification.Name("rivalCodeNotification")
+    static let twitterNameNotification = Notification.Name("twitterNameNotification")
+    
     private var hasInfo: Bool = false
+    
+    // Hold on to these until they are saved else where
+    private var userName: String = ""
+    private var rivalCode: String = ""
+    private var twitterName: String = ""
+    private var loginInfoViewController: UIViewController!
+    private var loginOptionsViewController: UIViewController!
+    private var signupStoryboard: UIStoryboard!
     @IBOutlet weak var signInButton: UIButton!
     @IBOutlet weak var optionContainer: UIView!
     
     override func viewDidLoad() {
+        self.view.addGestureRecognizer(UITapGestureRecognizer(target: self.view, action: #selector(UIView.endEditing(_:))))
+        NotificationCenter.default.addObserver(self, selector: #selector(onNotification(notification:)), name: LoginViewController.userNameNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(onNotification(notification:)), name: LoginViewController.rivalCodeNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(onNotification(notification:)), name: LoginViewController.twitterNameNotification, object: nil)
+        
+        self.signupStoryboard = UIStoryboard(name: "Signup", bundle: nil)
+        self.loginInfoViewController = self.signupStoryboard.instantiateViewController(identifier: "LoginInfoViewController")
+        self.loginOptionsViewController = self.signupStoryboard.instantiateViewController(identifier: "LoginOptionsViewController")
+        
         super.viewDidLoad()
     }
     
+    @objc func onNotification(notification:Notification){
+        if(notification.userInfo == nil) {
+            return
+        }
+        
+        if(notification.name == LoginViewController.userNameNotification) {
+            self.userName = notification.userInfo!["userName"] as! String
+            if (self.userName != "") {
+                signInButton.isEnabled = true
+            } else {
+                signInButton.isEnabled = false
+            }
+            
+        } else if(notification.name == LoginViewController.rivalCodeNotification) {
+            self.rivalCode = notification.userInfo!["rivalCode"] as! String
+        } else if(notification.name == LoginViewController.twitterNameNotification) {
+            self.twitterName = notification.userInfo!["twitterName"] as! String
+        }
+        
+        // Remove when comfortable with functionality
```

Comment:
```suggestion
        // FIXME Remove when comfortable with functionality
```
This makes finding debug-only code more searchable.